packages/spacecat-shared-rum-api-client/src/functions/traffic-acquisition.js

```diff
@@ -34,26 +35,35 @@ function transformFormat(trafficSources) {
     earned: value.earned,
     owned: value.owned,
     paid: value.paid,
+    maxTimeDelta: value.maxTimeDelta,
     sources: Object.entries(value)
-      .filter(([source]) => !MAIN_TYPES.includes(source))
+      .filter(([source]) => !MAIN_TYPES.includes(source) && source !== 'maxTimeDelta')
       .map(([source, views]) => ({ type: source, views })),
   }));
 }
 
 function formatTraffic(row) {
   const {
-    url, weight, type, category, vendor,
+    url, weight, type, category, vendor, events = [],
   } = row;
+
+  const timeDeltas = events.map((event) => event.timeDelta);
+  const maxTimeDelta = Math.max(...timeDeltas, 0);
```

Comment:
spread operator could be very costly for large arrays. Reduce can come handy here:

```
const maxTimeDelta = events.reduce((max, e) => Math.max(max, e.timeDelta), 0);
```